packages/lib/constants/url-regex.ts

```diff
@@ -0,0 +1,2 @@
+export const URL_REGEX =
+  /^(https?):\/\/(?:www\.)?[a-zA-Z0-9-]+\.[a-zA-Z0-9()]{2,}(?:\/[a-zA-Z0-9-._?&=/]*)?$/i;
```

Comment:
Use Zod for this, there's no need to have a regex :)

Reply:
Makes sense, I have added myself as a watcher for the issue in zod. Once that is resolved, I will come back to fix this 🤞🏻 

Here's a screen recording of us encountering the validation issue where I have demoed two cases,
- Using `.url()`
- Using Regex

[Redirect.webm](https://github.com/documenso/documenso/assets/23498248/74bc9a3d-40a0-4c90-b6e6-4ec8949de9e8)